Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp

```diff
@@ -375,6 +397,20 @@ static WPEView* wpeDisplayWaylandCreateView(WPEDisplay* display)
     return wpe_view_wayland_new(WPE_DISPLAY_WAYLAND(display));
 }
 
+static WPEInputMethodContext* wpeDisplayWaylandCreateInputMethodContext(WPEDisplay* display)
+{
+    auto* priv = WPE_DISPLAY_WAYLAND(display)->priv;
+    if (!priv->wlDisplay || !priv->wlCompositor)
+        return nullptr;
+
+    if (priv->textInputManagerV3)
+        return wpe_im_context_wayland_v3_new(WPE_DISPLAY_WAYLAND(display));
+    if (priv->textInputManagerV1)
+        return wpe_im_context_wayland_v1_new(WPE_DISPLAY_WAYLAND(display));
```

Comment:
This should just return the created context, not create a new one on every call, I think it's simpler if we have a single context attached to the display. In GTK things works differently because the IM context is a module loaded at runtime.

Reply:
we can have multiple im context per view with their own state that needs to be restored on focus